mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -233,14 +233,14 @@ bool AbstractControllerExecution::reachedGoalCheck()
 
 bool AbstractControllerExecution::cancel()
 {
-  cancel_ = true;
   // returns false if cancel is not implemented or rejected by the recovery behavior (will run until completion)
   if(!controller_->cancel())
   {
     ROS_WARN_STREAM("Cancel controlling failed or is not supported by the plugin. "
                         << "Wait until the current control cycle finished!");
     return false;
```

Comment:
We need the `cancel_ = true;` also here. Can you get the result of cancel() and return it after setting cancel_ to true?

Reply:
Ah right, this was before my first coffee.
Something like this?
```
bool AbstractControllerExecution::cancel()
{
  // returns false if cancel is not implemented or rejected by the recovery behavior (will run until completion)
  bool can_cancel = controller_->cancel();
  ROS_WARN_STREAM_COND(!can_cancel, "Cancel controlling failed or is not supported by the plugin. "
                                        << "Wait until the current control cycle finished!");
  cancel_ = true;
  return can_cancel;
}
```